lib/bbc-microbit-io.js

```diff
@@ -137,21 +137,27 @@ BBCMicrobitIO.prototype.digitalRead = function(pin, handler) {
 BBCMicrobitIO.prototype._onDiscover = function(microbit) {
   debug('on discover', microbit);
 
-  this._microbit = microbit;
+  //Certain Intel BT Chipsets seem to require a scan stop before connections are allowed, otherwise they'll throw a: Connection Rejected due to Limited Resources 
+  BBCMicrobit.stopScanning(() => {
```

Comment:
I think I see what's needed, we're not waiting for the stop scan to complete.

I think this change would be better to have in noble-device.

Change
```javascript
constructor.discover = function(callback) {
      var onDiscover = function(device) {
        constructor.stopDiscoverAll(onDiscover);

        callback(device);
      };

      callback._nobleDeviceOnDiscover = onDiscover;

      constructor.discoverAll(onDiscover);
    };
```

to something like

```javascript
constructor.discover = function(callback) {
      var onDiscover = function(device) {
        constructor.stopDiscoverAll(onDiscover, function() {
          callback(device);
        ).bind(this));
      };

      callback._nobleDeviceOnDiscover = onDiscover;

      constructor.discoverAll(onDiscover);
    };
```

We'll need to change `stopDiscoverAll` to include an additional callback parameter. What do you think? Do you have time to PR this?